src/WebJobs.Script/azurefunctions/functions.js

```diff
@@ -127,3 +128,27 @@ function getEntryPoint(f, context) {
 
     return f;
 }
+
+function stripFunctions(val) {
+    return stripFunctionsDepth(val, 2);
+}
+
+// strip functions from 'val' nested up to 'depth' within arrays and objects
+// ignores non-objects and buffers
+function stripFunctionsDepth(val, depth) {
```

Comment:
Debated on this. I figured that, as edge was doing all this unnecessary conversion under the covers anyway, we might as well move the stripping as high up in the pipeline as possible to cut the work as soon as possible.

However, that code looks super clean, I think I'm convinced just so that we keep all the conversion in one place. Also, as I think about this now, this only occurs in situations where we're converting to json which is a subset of functions.